src/components/Items/Item/Item.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {FC} from 'react';
+import {Ticket} from '../../../types/interfaces';
+import companies from '../../../constants/companies.json';
+
+
+import {Logo} from '../../../constants/logo'
+import {useTypedSelector} from "../../hooks/useTypedSelector";
+
+
+const Item: FC<Ticket> = ({id, price, companyId, segments}) => {
+
+
+    const Segments = useTypedSelector(state => state.segments);
+
+    // const price = (id: string) => {
+    //     return {...tickets.find(el => el.segments.includes(id))}.price
+    // }
+
+    const findSegment = () => {
+        return segments.map(segment => Segments.find(el => el.id == segment))
+    }
```

Comment:
[Сложность метода](www.youtube.com/watch?v=Fu4BzQNN0Qs) слишком большая. Можно оптимизировать с помощью Set:

```suggestion
    const findSegment = () => {
        const selectedSegmentsSet = new Set(segments);
        return Segments.find(el => selectedSegmentsSet.has(el.id));
    }
```

В твоем случае сложность будет O(n*m), в моём O(n+m); 
Т.е. если мы будем иметь по 100 элементов в каждом массиве, то в твоем случае в худшем случае мы получим **100x100 = 10,000** итераций, а в моем **100+100=200**.